src/include/sof/list.h

```diff
@@ -72,13 +72,24 @@ static inline void list_item_append(struct list_item *item,
 	list->prev = item;
 }
 
-/* delete item from the list */
+/* delete item from the list leaves deleted list item
+ *in undefined state list_is_empty won't return true
+ */
 static inline void list_item_del(struct list_item *item)
 {
 	item->next->prev = item->prev;
 	item->prev->next = item->next;
 }
 
+/* delete item from the list list iteam will be reinitialised
+ * and list_is_empty will return true
+ */
+static inline void list_item_del_init(struct list_item *item)
+{
+	list_item_del(item);
+	list_init(item);
```

Comment:
Here you init only the item not the list.
What do you mean by
> and list_is_empty will return true

Reply:
Item and list are interchangable.

> and list_is_empty will return true

When you just del item it still has pointers pointing to items in list it's not part of. When you init item it's just like newly created item that is not part of any list and will return true for list_is_empty - it doesn't check for null it next but if next == 'this item'